packages/components/src/form/FormInput.tsx

```diff
@@ -0,0 +1,200 @@
+import { pick, path } from 'ramda'
+import { isNotNil } from 'ramda-adjunct'
+import { useLayoutEffect } from 'react'
+import { useFormContext } from 'react-hook-form'
+
+import { cns } from '@stcland/utils'
+
+import { nullFormContext } from './Form'
+import Label from './FormLabel'
+
+//*****************************************************************************
+// Interface
+//*****************************************************************************
+export type FormInputStyles = Partial<{
+  root: React.HTMLAttributes<HTMLDivElement>['className'];
+  input: React.HTMLAttributes<HTMLInputElement>['className'];
+  label: React.HTMLAttributes<HTMLDivElement>['className'];
+  error: React.HTMLAttributes<HTMLDivElement>['className'];
+  info: React.HTMLAttributes<HTMLDivElement>['className'];
+}>;
+
+type TextAreaPropsShape = {
+  rows: number;
+};
+
+export type FormInputProps = {
+  name: string;
+  value?: any;
+  defaultValue?: any;
+  id?: string;
+  label?: string;
+  placeholder?: string;
+  valueAsNumber?: boolean;
+  required?: boolean;
+  disabled?: boolean;
+  hidden?: boolean;
+  touched?: boolean;
+  dirty?: boolean;
+  onFocus?: () => void;
+  textAreaProps?: TextAreaPropsShape;
+  customStyles?: Partial<FormInputStyles>;
+  onChange?: () => void;
+};
+
+export const formInputDefaultProps = {
+  label: '',
+  placeholder: '',
+  valueAsNumber: false,
+  required: false,
+  readOnly: false,
+  disabled: false,
+  hidden: false,
+  touched: false,
+  dirty: false,
+  onFocus: function () {},
+  textAreaProps: {},
+}
```

Comment:
@ianmarquez should we delete this variable? I don't see that we are using it anywhere, maybe an oversight? considering that we pass the defaults in the component signature.